### DISCUSSION TITLE
Blob size exceeds limit

### DISCUSSION THREAD

**Discussion by Johnite** [uber/cadence#4602] [Q&A]
I got a warn info:

2021-10-26T11:17:06.851+0800	warn	**Blob size exceeds limit.**	{"service": "cadence-frontend", "wf-domain-id": "30a01162-e822-41d2-9b40-9636e0dbce4b", "wf-id": "clickhouse_sql_user:realtime:enterShopCnt_24h", "wf-run-id": "14c5f9f5-89f1-4e40-bf3c-c5c6c15faeae", "wf-size": 474507, "blob-size-violation-operation": "RespondActivityTaskCompleted", "logging-call-at": "util.go:460"}

I think it may be controled by a dynamic config, but I dont know the argument.

**Comment by longquanzheng** [comment]
Blob size is the size of a single history event. This applies to any event input, like start workflow event, start activity event, or signal event.  Practically it is the size of input/output of workflow/activity. 

The size is greater than `BlobSizeLimitError` server will reject the request.

See https://pkg.go.dev/github.com/uber/cadence@v0.22.0/common/dynamicconfig#Key

search for `BlobSizeLimitWarn` and `BlobSizeLimitError`

Warn is the threshold that emitting warn log and metrics. Error is the threshold that will reject the requests.

**Comment by meiliang86** [comment]
@Johnite While you can update the config, it usually is not a good idea storing large blobs in your Cadence primary storage (cassandra/MySQL/etc).
If you intend to pass large blobs, it would be better if you can store the payload to some blob storage, and only pass references between your workflows and activities.
